numpy/ma/core.py

```diff
@@ -5213,7 +5213,8 @@ def round(self, decimals=0, out=None):
             out.__setmask__(self._mask)
         return out
 
-    def argsort(self, axis=None, kind='quicksort', order=None, fill_value=None):
+    def argsort(self, axis=-1, kind='quicksort', order=None,
+                endwith=True, fill_value=None):
```

Comment:
I don't see a way we could deprecate the axis argument value besides adding an argsort2 and deprecating the old one.

sort has a different order? that is unfortunate ...
I guess we could try to get away with that change, usage of positional arguments are probably not very common in the wild.